src/screens/IndexScreen.js

```diff
@@ -149,7 +151,7 @@ export const IndexScreen = ({ navigation }) => {
                 query: getFetchMoreQuery(query),
                 variables: {
                   ...queryVariables,
-                  offset: listItems.length
+                  offset: data?.[query]?.length
```

Comment:
why was this necessary and `listItems` changed with `data?.[query]?`? can the length of the one differ from the other?

Reply:
this length can now differ in the case of generic items, where we now filter the listitems for the dates.

this should have done `data.query.length` from the start, because that is what actually defines the offset.

Reply:
the list here shows `listItems` and fetches more data on scroll. so the offset needs to be the number of list items rendered. `listItems` is the number we see on the screen, right? and `data[query]` has a number returned by the server, which gets filtered before rendering, or have i misunderstood? so `data[query]` can have more entries than `listItems`, is this correct? and when refetching, we need to apply the number of already fetched data, because when applying the filtered (smaller) number, we could run into repeating items?

Reply:
if the server gives already filtered data for generic items as well, `data[query].length` would equal `listItems.lengths` again in all cases, right?

Reply:
yes.

the listItems length was previously always the same, but does not necessarily need to correspond to the amount of fetched data items. (in theory we could render multiple items per fetched dataEntry as well)
so we should always used the amount we fetched/received previously as the offset for the new fetch.

It could not only cause duplicate items, but also cause infinite refetching and appending as well, for example:
we fetch 10 items per fetchMore and only the first ever item gets not filtered.
that way we would always fetch with offset 1 without ever reaching the end of the available data.
and if item 12 was be valid again it would also never be shown either.